aws/resource_aws_glue_dev_endpoint.go

```diff
@@ -0,0 +1,588 @@
+			},
+
+			"failure_reason": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceAwsGlueDevEndpointCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).glueconn
+
+	var name string
+	if v, ok := d.GetOk("name"); ok {
+		name = v.(string)
+	} else {
+		name = resource.UniqueId()
+	}
+
+	input := &glue.CreateDevEndpointInput{
+		EndpointName: aws.String(name),
+		RoleArn:      aws.String(d.Get("role_arn").(string)),
+		Tags:         keyvaluetags.New(d.Get("tags").(map[string]interface{})).IgnoreAws().GlueTags(),
+	}
+
+	if kv, ok := d.GetOk("arguments"); ok {
+		arguments := make(map[string]string)
+		for k, v := range kv.(map[string]interface{}) {
+			arguments[k] = v.(string)
+		}
+		input.Arguments = aws.StringMap(arguments)
+	}
+
+	if v, ok := d.GetOk("extra_jars_s3_path"); ok {
+		input.ExtraJarsS3Path = aws.String(v.(string))
+	}
+
+	if v, ok := d.GetOk("extra_python_libs_s3_path"); ok {
+		input.ExtraPythonLibsS3Path = aws.String(v.(string))
+	}
+
+	if v, ok := d.GetOk("glue_version"); ok {
+		input.GlueVersion = aws.String(v.(string))
+	}
+
+	if v, ok := d.GetOk("number_of_nodes"); ok {
+		input.NumberOfNodes = aws.Int64(int64(v.(int)))
+	}
+
+	if v, ok := d.GetOk("number_of_workers"); ok {
+		input.NumberOfWorkers = aws.Int64(int64(v.(int)))
+	}
+
+	if v, ok := d.GetOk("public_key"); ok {
+		input.PublicKey = aws.String(v.(string))
+	}
+
+	if v, ok := d.GetOk("public_keys"); ok {
+		publicKeys := expandStringSet(v.(*schema.Set))
+		input.PublicKeys = publicKeys
+	}
+
+	if v, ok := d.GetOk("security_configuration"); ok {
+		input.SecurityConfiguration = aws.String(v.(string))
+	}
+
+	if v, ok := d.GetOk("security_group_ids"); ok {
+		securityGroupIDs := expandStringSet(v.(*schema.Set))
+		input.SecurityGroupIds = securityGroupIDs
+	}
+
+	if v, ok := d.GetOk("subnet_id"); ok {
+		input.SubnetId = aws.String(v.(string))
+	}
+
+	if v, ok := d.GetOk("worker_type"); ok {
+		input.WorkerType = aws.String(v.(string))
+	}
+
+	log.Printf("[DEBUG] Creating Glue Dev Endpoint: %#v", *input)
+	err := resource.Retry(1*time.Minute, func() *resource.RetryError {
+		_, err := conn.CreateDevEndpoint(input)
+		if err != nil {
+			// Retry for IAM eventual consistency
+			if isAWSErr(err, glue.ErrCodeInvalidInputException, "should be given assume role permissions for Glue Service") {
+				return resource.RetryableError(err)
+			}
+			if isAWSErr(err, glue.ErrCodeInvalidInputException, "S3 endpoint and NAT validation has failed for subnetId") {
+				return resource.RetryableError(err)
+			}
+
+			return resource.NonRetryableError(err)
+		}
+		return nil
+	})
+
+	if isResourceTimeoutError(err) {
+		_, err = conn.CreateDevEndpoint(input)
+	}
+
+	if err != nil {
+		return fmt.Errorf("error creating Glue Dev Endpoint: %s", err)
+	}
+
+	d.SetId(name)
+
+	log.Printf("[DEBUG] Waiting for Glue Dev Endpoint (%s) to become available", d.Id())
+	stateConf := &resource.StateChangeConf{
```

Comment:
now that the refresh function is extracted, lets extract this waiter as well to the glue waiter package(see other glue functions in that package)